client/widgets/markets/TradePanels.h

```diff
@@ -28,19 +28,19 @@ class CTradeableItem : public SelectableSlot, public std::enable_shared_from_thi
 	using ClickPressedFunctor = std::function<void(const std::shared_ptr<CTradeableItem>&)>;
 
 	EType type;
-	int id;
-	const int serial;
+	int32_t id;
+	const int32_t serial;
```

Comment:
Not sure if it worth the effort, but you could use `VariantIdentifier` here.
Check TradeItemSell / TradeItemBuy defines for example.

Reply:
I assume that CTradeableItem will not have an id field at all. It will be in CComponentHolder. But that will be later.